### PR TITLE
Add Monday provider

### PR DESCRIPTION
**Pull request by amiranagram**
Add Socialite provider for Monday:

https://apps.developer.monday.com/docs/oauth

### CODE REVIEW

File: src/Monday/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+You should now be able to use the provider like you would regularly use Socialite (assuming you have the facade installed):
+
+```php
+return Socialite::driver('monday')->redirect();
+```
+
+### Returned User fields
+
+- `birthday`
```
Comment (atymic) [inline src/Monday/README.md:46]: Maybe worth noting that most of these are not set on the object, but you need to getRaw?

**Review by atymic — COMMENTED** [review]
Looks good, one comment.

File: src/Monday/Provider.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->httpClient
+            ->post('https://api.monday.com/v2', [
+                'headers' => [
```
Comment (lucasmichot) [inline src/Monday/Provider.php:56]: Please use `RequestOptions::HEADERS` here

File: src/Monday/Provider.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    'Content-Type'  => 'application/json',
+                    'Authorization' => $token,
+                ],
+                'body' => json_encode([
```
Comment (lucasmichot) [inline src/Monday/Provider.php:60]: Same here, `RequestOptions::BODY`

**Review by lucasmichot — CHANGES_REQUESTED** [review]
One small change required

File: src/Monday/Provider.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+namespace SocialiteProviders\Monday;
+
+use Laravel\Socialite\Two\AbstractProvider;
+use SocialiteProviders\Manager\ConfigTrait;
+use SocialiteProviders\Manager\Contracts\OAuth2\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    use ConfigTrait;
```
Comment (lucasmichot) [inline src/Monday/Provider.php:12]: Remove this trait import

File: src/Monday/Provider.php [hunk 5]
```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace SocialiteProviders\Monday;
+
+use Laravel\Socialite\Two\AbstractProvider;
+use SocialiteProviders\Manager\ConfigTrait;
+use SocialiteProviders\Manager\Contracts\OAuth2\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
```
Comment (lucasmichot) [inline src/Monday/Provider.php:10]: No need to implements `ProviderInterface`

**Review by lucasmichot — DISMISSED** [review]
Some small changes required

**Comment by amiranagram** [comment]
@lucasmichot updated.

**Review by atymic — APPROVED** [review]

**Review by atymic — CHANGES_REQUESTED** [review]
@amiranagram looks like the syntax doesn't work for php 7.2.
Just force php 7.3 or higher in the composer json please

**Comment by amiranagram** [comment]
@atymic can you run it again, I fixed the NOWDOC?

**Comment by amiranagram** [comment]
Hi @atymic @lucasmichot when can we get this provider as a composer package?

**Pull request merged**